include/pdmatrix.hh

```diff
@@ -34,13 +37,23 @@ public:
 
   auto operator+=(const Matrix<T> &m) -> Matrix<T> * {
     if (!(this->rows() == m.rows() && this->cols() == m.cols()))
-      throw "matrices have different size";
+      throw std::invalid_argument("matrices have different size");
     for (auto i = matrix_.begin(), j = m.matrix_.begin(); i != matrix_.end();
          i++, j++) {
       *i = *i + *j;
     }
     return this;
   }
+
+  auto operator-=(const Matrix<T> &m) -> Matrix<T> * {
+    if (!(this->rows() == m.rows() && this->cols() == m.cols()))
+      throw std::invalid_argument("matrices have different size");
+    for (auto i = matrix_.begin(), j = m.matrix_.begin(); i != matrix_.end();
+         i++, j++) {
```

Comment:
It is advised to use preincrement instead of postincrement when working with iterators or indices. When the return value is ignored, the pre ++i is never less efficient than the post i++.